misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/DevelopmentJobQueueTest.kt

```diff
@@ -0,0 +1,55 @@
+package misk.jobqueue
+
+import misk.MiskTestingServiceModule
+import misk.inject.KAbstractModule
+import misk.logging.LogCollectorModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import wisp.logging.LogCollector
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+internal class DevelopmentJobQueueTest {
+  @MiskTestModule private val module = DevelopmentTestModule()
+
+  @Inject private lateinit var exampleJobEnqueuer: ExampleJobEnqueuer
+  @Inject private lateinit var logCollector: LogCollector
+
+  @Test
+  fun basic() {
+    exampleJobEnqueuer.enqueueRed("stop sign")
+    assertThat(logCollector.takeMessages(ExampleJobHandler::class)).containsExactlyInAnyOrder(
```

Comment:
I think I just encountered a test flake with this test suite https://github.com/cashapp/misk/actions/runs/4558183489/jobs/8040758315

(the "test report" output is frustratingly lacking in an actual stack trace that I can find for the specific failure)

Enqueue and then immediate assertion of job handling seems prone to flakiness since that assertion could run (and fail) before the job had a chance to be handled. Perhaps better to have the job handler itself decrement a `CountDownLatch` or similar and gate our assertions against that.